lib/sles4sap/ipaddr2.pm

```diff
@@ -166,22 +197,47 @@ sub ipaddr2_azure_deployment {
             name => ipaddr2_azure_storage_account());
     }
 
+    # If required, create on the fly the cloud-init script
+    my $cloud_init_file;
+    if ($args{cloudinit}) {
+        my $register_str = "";
+        if ($args{scc_code}) {
+            $register_str = <<END;
+
+bootcmd:
+  - registercloudguest --clean
+  - registercloudguest --force-new -r $args{scc_code}
+END
+        }
+
+        my $cloud_init_content = <<END;
+#cloud-config
+package_upgrade: false${register_str}
+packages:
+  - nginx
+runcmd:
+  - 'echo "I am \$(hostname)" > /srv/www/htdocs/index.html'
+  - sudo systemctl enable nginx.service
+  - sudo systemctl start nginx.service
+END
```

Comment:
I think the way this is constructed is a bit confusing. Why not something like:
```
if ($args{cloudinit}) {
    my $cloud_init_content = <<END;
#cloud-config
package_upgrade: false
packages:
  - nginx
runcmd:
  - 'echo "I am \$(hostname)" > /srv/www/htdocs/index.html'
  - sudo systemctl enable --now nginx.service

END

    if ($args{scc_code}) {
        $cloud_init_content .= <<END
bootcmd:
  - registercloudguest --clean
  - registercloudguest --force-new -r $args{scc_code}
END
    }
}
```
After all, it looks like cloud-init config is YAML and `bootcmd` is not a sub-element of anything else. Or am I misunderstanding?

Reply:
I'm not sure if the order matter or not, but in any case I'll give it a try to your suggestion as code is much cleaner